dashboard/app/controllers/plc/user_course_enrollments_controller.rb

```diff
@@ -16,7 +16,11 @@ def create
     if @user_course_enrollment.save
```

Comment:
you could also use find_or_create here and save a few steps. create_enrollment and the exists check below could be one line (not work it just for the one-line-ness but maybe worth it to not repeat the search criteria)
````
user = User.find_by_email_or_hashed_email(user_course_enrollment_params[:user_email])
@user_course_enrollment = UserCourseEnrollment.find_or_create_by(user: user, plc_course_id: u_c_e_params[:plc_course_id])

if @user_course_enrollment.valid? # found it or created and saved
  redirect_to action: :index
else # attempted to save but caused validation errors
  redirect_to action: :new
end
````